src/server/pps/server/api_server.go

```diff
@@ -2641,9 +2590,21 @@ func (a *apiServer) listPipelinePtr(ctx context.Context,
 			}
 			// Get parent commit
 			pachClient := a.env.GetPachClient(ctx)
-			ci, err := pachClient.InspectCommit(ppsconsts.SpecRepo, p.SpecCommit.Branch.Name, p.SpecCommit.ID)
+			// allow reading spec commit regardless of calling user
+			pachClient.SetAuthToken(p.AuthToken)
+			ci, err := pachClient.PfsAPIClient.InspectCommit(
+				pachClient.Ctx(),
+				&pfs.InspectCommitRequest{
+					Commit:     p.SpecCommit,
+					BlockState: pfs.CommitState_STARTED,
+				},
+			)
 			switch {
 			case err != nil:
+				err = grpcutil.ScrubGRPC(err)
+				if auth.IsErrNoRoleBinding(err) {
+					return nil // a missing role binding suggests the spec repo is gone
+				}
```

Comment:
This doesn't seem like a great way to address this, I guess I'd like `InspectCommit` to give `ErrRepoNotFound` in this case, rather than failing in auth, but I'm not sure if that should be a change in `Authorize` itself or special case logic for commits.

Reply:
I'll make an issue to clean this up - it's not a good user experience anywhere.